kotlinlib/src/mill/kotlinlib/ksp/KspModule.scala

```diff
@@ -0,0 +1,157 @@
+package mill.kotlinlib.ksp
+
+import mill.*
+import mill.api.{PathRef, Result}
+import mill.define.Task
+import mill.kotlinlib.worker.api.{KotlinWorker, KotlinWorkerTarget}
+import mill.kotlinlib.{Dep, DepSyntax, KotlinModule}
+
+import java.io.File
+
+/**
+ * Sets up the kotlin compiler for using KSP (Kotlin Symbol Processing)
+ * by plugging in the symbol-processing and symbol-processing-api dependencies.
+ *
+ * Use of kotlin-compiler-embedded is also recommended (and thus enabled by default)
+ * to avoid any classpath conflicts between the compiler and user defined plugins!
+ */
+trait KspModule extends KotlinModule {
+
+  def symbolProcessingVersion: T[String]
+
+  /**
+   * Mandatory plugins that are needed for KSP to work.
+   * For more info go to [[https://kotlinlang.org/docs/ksp-command-line.html]]
+   *
+   * @return
+   */
+  def kspPlugins: T[Agg[Dep]] = Task {
+    Agg(
+      ivy"com.google.devtools.ksp:symbol-processing-api:${kotlinVersion()}-${symbolProcessingVersion()}",
+      ivy"com.google.devtools.ksp:symbol-processing:${kotlinVersion()}-${symbolProcessingVersion()}"
+    )
+  }
+
+  def kspProjectBaseDir: T[PathRef] = Task {
+    PathRef(moduleDir)
+  }
+
+  /** KSP output dir */
+  def kspOutputDir: T[PathRef] = Task {
+    PathRef(T.dest / "generated" / "ksp" / "main")
+  }
```

Comment:
Why is this needed. I results in a design we try to discourage in Mill. A task should not write into the dest of another task. Since this task is only used from the `generateSourcesWithKSP` task, why it it externalized? Instead, the `generateSourcesWithKSP` task should manage that dir by itself. If later, other tasks need access to it, them may as well just refer to that task.

Reply:
yes, I need to group these better, they are all part of the ksp domain, I'll come back with something better

Reply:
I've removed all of these and have them locally defined in the `generateSourcesWithKSP` task